hail/python/hailtop/batch/job.py

```diff
@@ -877,6 +877,19 @@ async def _compile(self, local_tmpdir, remote_tmpdir, *, dry_run=False):
         return True
 
 
+UnpreparedArg = '_resource.ResourceType' | List['UnpreparedArg'] | Tuple['UnpreparedArg', ...] | Dict[str, 'UnpreparedArg'] | Any
```

Comment:
I think we have to wait until [3.10](https://peps.python.org/pep-0604/) for this unfortunately.